transforms/src/button-v0-to-v1.ts

```diff
@@ -0,0 +1,110 @@
+import { Transform, JSCodeshift, FileInfo, API, Options, ImportDeclaration, ASTPath, JSXElement } from 'jscodeshift';
+
+export const transform: Transform = (fileInfo: FileInfo, api: API, options: Options) => {
+  const j: JSCodeshift = api.jscodeshift;
+
+  const printOptions = options.printOptions || {
+    printWidth: 140,
+    quote: 'single',
+    tabWidth: 2,
+    trailingComma: true,
+  };
+
+  const root = j(fileInfo.source);
+  root
+    // Find all imports that import from '@fluentui/react-native' or '@fluentui-react-native/button'
+    .find(
+      j.ImportDeclaration,
+      (path: ImportDeclaration) => path.source.value === '@fluentui/react-native' || path.source.value === '@fluentui-react-native/button',
+    )
+    .forEach((path: ASTPath<ImportDeclaration>) => {
+      if (!path.value.specifiers) {
+        return;
+      }
+
+      // Find the first import specifier that imports Button, PrimaryButton or StealthButton.
+      const buttonSpecifier = path.value.specifiers.findIndex(
+        (specifier) =>
+          specifier.type === 'ImportSpecifier' &&
+          (specifier.imported.name === 'Button' ||
+            specifier.imported.name === 'PrimaryButton' ||
+            specifier.imported.name === 'StealthButton'),
+      );
+
+      // Change that import to be 'ButtonV1 as Button'
+      if (buttonSpecifier !== undefined && buttonSpecifier !== -1) {
+        path.value.specifiers[buttonSpecifier] = j.importSpecifier(j.identifier('ButtonV1'), j.identifier('Button'));
+      }
+
+      // Filter out all other instances of importing Button, PrimaryButton, or StealthButton so that ButtonV1 is only imported once.
+      path.value.specifiers = path.value.specifiers.filter(
+        (specifier) =>
+          specifier.type !== 'ImportSpecifier' ||
+          (specifier.imported.name !== 'Button' &&
+            specifier.imported.name !== 'PrimaryButton' &&
+            specifier.imported.name !== 'StealthButton'),
+      );
+    });
+
+  // Change JSX instances of previous buttons to new type of button
+  root.findJSXElements('PrimaryButton').forEach((path: ASTPath<JSXElement>) => {
+    if (path.value.openingElement.name.type === 'JSXIdentifier') {
+      path.value.openingElement.name.name = 'Button';
```

Comment:
Would this catch a component made with compose that has a Button slot?

Reply:
Can you give a code example of what you're thinking of? In general no, this script doesn't handle any compose/customize scenarios

Reply:
I guess that's not a client scenario. nvm.